src/main/java/nl/ou/debm/common/feature1/LoopProducer.java

```diff
@@ -554,7 +554,16 @@ private void getDummyStatements(int currentDepth, StatementPrefs prefs, List<Str
             // function call or dummy?
             if (Misc.rnd.nextDouble() < DBLCHANCEOFFUNCTIONCALLASDUMMY){
                 // function call
-                m_cgenerator.getFunction(currentDepth);
+                var function = m_cgenerator.getFunction(currentDepth, m_cgenerator.getRawDataType());
+                if(function.getParameters().isEmpty() && !function.hasVarArgs()) {
+                    list.add(function.getName() + "();");
+                }else{
+                    var arguments = new ArrayList<String>();
+                    for(var parameter : function.getParameters())
+                        arguments.add(m_cgenerator.getNewExpression(currentDepth+1, parameter.getType()));
+
+                    list.add(function.getName() + "(" + String.join(", ", arguments) + ");");
+                }
             }
```

Comment:
I don't get the if-structure.
First you test empty argument list and no var args. Makes sense: no arguments at all, so be done.
But the getParameters in the else-part may also return an empty list (in case empty parameters list and var args).
Conclusion 1: you don't need to test for no arguments, as the String.join will return an empty string if the list is empty. 
Conclusion 2: hasVarArgs will do nothing now, because the arguments are only inferred from function.getParameters.
I would rephrase:
```suggestion
if (function.hasVarArgs { 
  // process varargs
}
else{
// keep the else you've got at the moment, as it will both work on no, or 1+ arguments
}```